lib/plugins/executable/index.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const fs = require('fs');
+const os = require('os');
+const streamPromise = require('stream-promise');
+const fse = BbPromise.promisifyAll(require('fs-extra'));
+const isStandaloneExecutable =
+  require('../../utils/isStandaloneExecutable') && process.platform !== 'win32';
+const currentVersion = require('../../../package').version;
+const fetch = require('node-fetch');
+
+const BINARIES_DIR_PATH = `${os.homedir()}/.serverless/bin`;
+const BINARY_TMP_PATH = `${BINARIES_DIR_PATH}/serverless-tmp`;
+const BINARY_PATH = `${BINARIES_DIR_PATH}/serverless`;
+
+module.exports = class Executable {
+  constructor(serverless) {
+    this.serverless = serverless;
+
+    this.commands = {
+      upgrade: {
+        isHidden: !isStandaloneExecutable,
+        usage: 'Upgrade Serverless',
+        lifecycleEvents: ['upgrade'],
+      },
+      uninstall: {
+        isHidden: !isStandaloneExecutable,
+        usage: 'Uninstall Serverless',
+        lifecycleEvents: ['uninstall'],
+      },
+    };
+
+    this.hooks = {
+      'upgrade:upgrade': () => {
+        return isStandaloneExecutable ? this.upgrade() : this.rejectCommand('upgrade');
+      },
+      'uninstall:uninstall': () => {
+        return isStandaloneExecutable ? this.uninstall() : this.rejectCommand('uninstall');
+      },
+    };
+  }
+
+  upgrade() {
+    return fetch('https://api.github.com/repos/serverless/serverless/releases/latest')
```

Comment:
Not sure about their throttling, but we might run into issues if the user has queried the GitHub API too often. Not sure if there's anything specific we can do about this ATM.

Reply:
Yes, I wouldn't worry about that, as it's a one time command. So trying to bake in support for `GITHUB_TOKEN` seems overkill (we may address it, when learn that there's indeed a demand for it)

Reply:
Yes, that was what I was thinking as well...